http/handler.go

```diff
@@ -139,6 +141,18 @@ func requestAuth(r *http.Request, req *logical.Request) *logical.Request {
 	return req
 }
 
+// Determines the type of the error being returned and sets the HTTP
```

Comment:
Only the latest error type will be considered. In order for this to work as expected, we should follow a practice, i.e. when `err != nil`, we should *always* do `return err` instead of `return fmt.Errorf("err:%s",err)`. This way, the error type once set, will retain it's type all the way back into this method.

Since we were only generating internal error till date, I guess it should be fine if we start this practice now. This might not affect any of the existing use-cases.

The status code is not being set anywhere after any call crosses core, except `logical.HTTPStatusCode` which is a different code flow.

In case of multiple errors using multierror, the default case will be `StatusInternalServerError`. It makes sense as well. There will be more than one error and we'll not know which status to set (unless there are predefined priorities to status codes, are there?).

Also, since multierror implements error interface, user will get to see the all the errors in a formatted manner, and hopefully not complain for the proper status code.